autogen/agentchat/contrib/retrieve_user_proxy_agent.py

```diff
@@ -244,18 +284,95 @@ def retrieve_docs(self, problem: str, n_results: int = 20, search_string: str =
         self._doc_idx = -1  # the index of the current used doc
-        self._results = {}  # the results of the current query
+        self._results = []  # the results of the current query
         self._intermediate_answers = set()  # the intermediate answers
         self._doc_contents = []  # the contents of the current used doc
         self._doc_ids = []  # the ids of the current used doc
         self._current_docs_in_context = []  # the ids of the current context sources
         self._search_string = ""  # the search string used in the current query
+        self._distance_threshold = self._retrieve_config.get("distance_threshold", -1)
         # update the termination message function
         self._is_termination_msg = (
             self._is_termination_msg_retrievechat if is_termination_msg is None else is_termination_msg
         )
+        if isinstance(self._vector_db, str):
+            if not isinstance(self._db_config, dict):
+                raise ValueError("`db_config` should be a dictionary.")
+            if "embedding_function" in self._retrieve_config:
+                self._db_config["embedding_function"] = self._embedding_function
+            self._vector_db = VectorDBFactory.create_vector_db(db_type=self._vector_db, **self._db_config)
         self.register_reply(Agent, RetrieveUserProxyAgent._generate_retrieve_user_reply, position=2)
 
+    def _init_db(self):
+        if not self._vector_db:
+            return
+
+        IS_TO_CHUNK = False  # whether to chunk the raw files
+        if self._new_docs:
+            IS_TO_CHUNK = True
+        if not self._docs_path:
+            try:
+                self._vector_db.get_collection(self._collection_name)
+                logger.warning(f"`docs_path` is not provided. Use the existing collection `{self._collection_name}`.")
+                self._overwrite = False
+                self._get_or_create = True
+                IS_TO_CHUNK = False
+            except ValueError:
+                raise ValueError(
+                    "`docs_path` is not provided. "
+                    f"The collection `{self._collection_name}` doesn't exist either. "
+                    "Please provide `docs_path` or create the collection first."
+                )
+        elif self._get_or_create and not self._overwrite:
+            try:
+                self._vector_db.get_collection(self._collection_name)
+                logger.info(f"Use the existing collection `{self._collection_name}`.", color="green")
+            except ValueError:
+                IS_TO_CHUNK = True
+        else:
+            IS_TO_CHUNK = True
+
+        self._vector_db.active_collection = self._vector_db.create_collection(
+            self._collection_name, overwrite=self._overwrite, get_or_create=self._get_or_create
+        )
+
+        docs = None
+        if IS_TO_CHUNK:
+            if self.custom_text_split_function is not None:
+                chunks = split_files_to_chunks(
```

Comment:
This should be:
`chunks, sources = split_files_to_chunks`

Otherwise chunks will contain the Tuple(chunks, sources).